### ISSUE TITLE
AbstractElementTagProcessor not matching element name

### ISSUE THREAD

**Issue by sofiageo** [thymeleaf/thymeleaf#575]
I'm trying to create some custom thymeleaf tags but I'm having trouble making AbstractElementTagProcessor doProcess to work with a normal XML element e.g: 

`<customtag>body</customtag>`

I've followed the tutorial for extending thymeleaf, tried everything but nothing seems to match my element. This looks to be an issue on thymeleaf 3.0.2 and 3.0.3.

It only seems to be matching when extending the AbstractAttributeTagProcessor class and using an attribute name to match the tag, and **only** if I use a constructor with null element name e.g:

`<div customparam="1">body</div>`

`super(templateMode, dialectPrefix, null, false, ATTR_NAME, false, PRECEDENCE, true);`

**Comment by danielfernandez** [comment]
Are you applying a prefix to your dialect? And if so... are you including such prefix in your markup, like `<prefix:customtag>`? 

Also, are you making sure that you are adding your processor to the custom dialect definition? 

Could you please post what the constructor for you element (not attribute) processor looks like?

**Comment by sofiageo** [comment]
Hi, thanks for your quick response and sorry for not providing many details. I've tried both with prefix and without prefix. I'm including the prefix on my markup, and I've also tried without a prefix. I've added both the custom processor and the StandardXmlNsTagProcessor processor to the custom dialect definition. I've also tried only with the custom processor on the custom dialect. 

The constructor for the element processor is almost the same as the one I'm using for the attribute processor:
`super(templateMode, dialectPrefix, ELEM_NAME, true, null, false, PRECEDENCE);`

dialect prefix was: "perm" and ELEM_NAME was "haspermission" in this example. While debugging, I can see the processor having a MatchingName of perm:haspermission, but it never matches in runtime.

**Comment by danielfernandez** [comment]
I've not been able to replicate this. Could you please make sure that the template engine you are setting into your processors actually matches the template engine used for processing your template?

And if so, could you please provide some code, ideally a test project that shows this behaviour?

Thanks

**Comment by sofiageo** [comment]
I will try to create a test project later in the evening and post it here. If you can't reproduce it though, I doubt I will be able to. Maybe it's my Java version (8) or some of my project dependencies that are messing with this. If you can tell me what Class / method does the actual matching, I can try to debug it and provide more details.

**Comment by danielfernandez** [comment]
Have you checked your dialect's processors are configured to work for the same template mode you are using?

**Comment by danielfernandez** [comment]
@gsf-greece where you finally able to create some code that reproduces this that we can have a look at?

**Comment by sofiageo** [comment]
Hi, sorry I was extremely busy / tired last week. I'll try to reproduce this today in a new project.

**Comment by sofiageo** [comment]
I created a new project with the same code for custom tags, and it works fine. I then tested the main project that had the original issues, and it also works fine. I don't know what may have trigger this in the first place. 
I will keep a close eye on it (since I now know it should work with this configuration), and I will try to pinpoint the cause before opening a ticket in the future. Thanks for your time and help!
